cdap-gateway/src/main/java/co/cask/cdap/gateway/router/handlers/IdleEventProcessor.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.gateway.router.handlers;
+
+import org.jboss.netty.channel.Channel;
+import org.jboss.netty.channel.ChannelHandlerContext;
+import org.jboss.netty.channel.ChannelStateEvent;
+import org.jboss.netty.channel.MessageEvent;
+import org.jboss.netty.handler.codec.http.HttpChunk;
+import org.jboss.netty.handler.codec.http.HttpRequest;
+import org.jboss.netty.handler.codec.http.HttpResponse;
+import org.jboss.netty.handler.timeout.IdleState;
+import org.jboss.netty.handler.timeout.IdleStateAwareChannelHandler;
+import org.jboss.netty.handler.timeout.IdleStateEvent;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Handles states when the router has been idle for a configured time interval, by closing the channel if an
+ * HTTP Request is not in progress.
+ */
+public class IdleEventProcessor extends IdleStateAwareChannelHandler {
+  private static final Logger LOG = LoggerFactory.getLogger(IdleEventProcessor.class);
+  private volatile boolean requestInProgress;
```

Comment:
right. removed `volatile`.
http://stackoverflow.com/questions/8304309/keeping-state-in-a-netty-channelhandler